Sources/Commands/SwiftPackageTool.swift

```diff
@@ -863,6 +865,138 @@ extension SwiftPackageTool {
+                case .sourceCodeFormatting:
+                    return command == "format-source-code"
+                case .custom(let verb, _):
+                    return command == verb
+                }
+            }
+        }
+
+        func run(_ swiftTool: SwiftTool) throws {
+            // Load the workspace and resolve the package graph.
+            let packageGraph = try swiftTool.loadPackageGraph()
+            
+            // Find the plugins that match the command.
+            let matchingPlugins = findPlugins(matching: command, in: packageGraph)
+
+            // Complain if we didn't find exactly one.
+            if matchingPlugins.isEmpty {
+                swiftTool.observabilityScope.emit(error: "No plugins found for '\(command)'")
+                throw ExitCode.failure
+            }
+            else if matchingPlugins.count > 1 {
+                swiftTool.observabilityScope.emit(error: "\(matchingPlugins.count) plugins found for '\(command)'")
+                throw ExitCode.failure
+            }
+            
+            // At this point we know we found exactly one command plugin, so we run it.
+            let plugin = matchingPlugins[0]
+            print("Running plugin \(plugin)")
+            
+            // Find the targets (if any) specified by the user.
+            var targets: [String: ResolvedTarget] = [:]
+            for target in packageGraph.allTargets {
+                if targetNames.contains(target.name) {
+                    if targets[target.name] != nil {
+                        swiftTool.observabilityScope.emit(error: "Ambiguous target name: ‘\(target.name)’")
+                        throw ExitCode.failure
+                    }
+                    targets[target.name] = target
+                }
+            }
+            assert(targets.count <= targetNames.count)
+            if targets.count != targetNames.count {
+                let unknownTargetNames = Set(targetNames).subtracting(targets.keys)
+                swiftTool.observabilityScope.emit(error: "Unknown targets: ‘\(unknownTargetNames.sorted().joined(separator: "’, ‘"))’")
+                throw ExitCode.failure
+            }
+
+            // Configure the plugin invocation inputs.
+
+            // The `plugins` directory is inside the workspace's main data directory, and contains all temporary files related to all plugins in the workspace.
+            let dataDir = try swiftTool.getActiveWorkspace().location.workingDirectory
+            let pluginsDir = dataDir.appending(component: "plugins")
+
+            // The `cache` directory is in the plugins directory and is where the plugin script runner caches compiled plugin binaries and any other derived information.
+            let cacheDir = pluginsDir.appending(component: "cache")
+            let pluginScriptRunner = DefaultPluginScriptRunner(cacheDir: cacheDir, toolchain: try swiftTool.getToolchain().configuration, enableSandbox: false)
```

Comment:
`enableSandbox: false` intentional?

Reply:
No, temporary.  I'll fix that as well.  Thanks.